oc_cdtapi/DmsGetverAPI.py

```diff
@@ -184,6 +317,88 @@ def get_dms_log_url(self, distr_id, distr_option):
 
         return url
 
+    def json_or_none(self, resp):
+        """
+        Returns requests response's json if status code is 200-299 or logs error and returns None
+        :param requests.response resp: requests' response
+        :return: requests.response.json or None
+        """
+        logging.debug('Reached json_or_none')
+        status_code = resp.status_code
+        if 200 <= status_code <= 299:
+            logging.debug('[%s] status code received, trying to return json' % status_code)
+            return resp.json()
+        else:
+            logging.error('[%s] status code received, assuming an error, returning None' % status_code)
+        return None
+
+    def login(self):
+        """
+        Obtain auth token
+        """
+        logging.debug('Reached login')
+        username = os.getenv('DBSM2_USER')
+        password = os.getenv('DBSM2_PASSWORD')
+        login_data = {
+            'grant_type': '',
+            'username': username,
+            'password': password,
+            'scope': '',
+            'client_id': '',
+            'client_secret': ''
+        }
+        rh = self.raise_exception_high
+        self.raise_exception_high = 499
+        url = posixpath.join('api', 'v1', 'auth', 'access-token')
+        try:
+            resp = self.post(url, data=login_data)
+            resp_data = resp.json()
+            if resp.status_code == 200:
+                token_type = resp_data['token_type']
+                access_token = resp_data['access_token']
+            else:
+                logging.debug('Server returned an error [%s] [%s]' % (resp.status_code, resp.text))
+                return None
+        except:
```

Comment:
I did not get the reason for `try` block if exception is to be raised immediately. This is the default.